src/api/client/Network.ts

```diff
@@ -549,4 +549,23 @@ export class Network {
   public async getMiddlewareMetadata(): Promise<MiddlewareMetadata | null> {
     return this.context.getMiddlewareMetadata();
   }
+
+  /**
+   * Get the number of blocks the middleware needs to process to be synced with chain
```

Comment:
I think we should add a comment here recommending a number. To be useful for user, they should know it shouldn't exceed 10 or 15 blocks.